it/test/uk/gov/hmrc/agentsubscriptionfrontend/connectors/AgentSubscriptionConnectorISpecIt.scala

```diff
@@ -379,6 +380,32 @@ class AgentSubscriptionConnectorISpecIt extends BaseISpecIt with MetricTestSuppo
     }
   }
 
+  "getAmlsSubscriptionRecord" should {
+    "return amls Subscription Record" in {
+      AgentSubscriptionStub.givenAmlsRecordFound("12345", Approved)
+      val result = await(connector.getAmlsSubscriptionRecord("12345"))
+
+      result shouldBe Some(
+        AmlsSubscriptionRecord(Approved, "111234567890123", Some(LocalDate.of(2021, Month.JANUARY, 1)), Some(LocalDate.now().plusDays(2)), None)
+      )
+    }
+
+    " return None when agent subscription record returns a 404 response" in {
+      AgentSubscriptionStub.givenAmlsRecordNotFound("12345")
+      val result = await(connector.getAmlsSubscriptionRecord("12345"))
+
+      result shouldBe None
+    }
+
+    "throw Upstream4xxResponse if details doesn't match records" in {
```

Comment:
Would you mind just rewording this, it doesn't sound accurate with the scenario. Looking at the connector code this scenario is: throw an UpstreamErrorResponse exception when agent subscription returns an unexpected status